docs/how-to/h-deploy-azure.md

```diff
@@ -0,0 +1,239 @@
+
+```shell
+juju version
+```
+
+which should show:
+
+```shell
+3.6-rc1-genericlinux-amd64
+```
+
+### Azure CLI
+
+Follow the [user guide](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli-linux?pivots=apt) for installing the Azure CLI on Linux distributions.
+
+Verify that it is correctly installed running the command below:
+
+```shell
+az --version
+```
+
+which should show the following output:
+
+```shell
+azure-cli                         2.65.0
+
+core                              2.65.0
+telemetry                          1.1.0
+
+Dependencies:
+msal                              1.31.0
+azure-mgmt-resource               23.1.1
+
+Python location '/opt/az/bin/python3'
+Extensions directory '/home/deusebio/.azure/cliextensions'
+
+Python (Linux) 3.11.8 (main, Sep 25 2024, 11:33:44) [GCC 11.4.0]
+
+Legal docs and information: aka.ms/AzureCliLegal
+
+
+Your CLI is up-to-date.
+```
+
+### Authenticate
+
+> For more information on how to authenticate, refer to the [Juju documentation](/t/the-microsoft-azure-cloud-and-juju/1086) and [a dedicated user guide on how to register Azure on Juju](/t/how-to-use-juju-with-microsoft-azure/15219).
+
+First of all, retrieve your subscription-id
+
+```shell
+az login
+```
+
+After providing the authentication via web browser, you will be shown a list of information and a table with the subscriptions connected to your user, e.g.:
+
+```shell
+
+...
+
+No     Subscription name                     Subscription ID                       Tenant
+-----  ------------------------------------  ------------------------------------  -------------
+[1] *  <subscription_name>                   <subscription_id>                     canonical.com
+[2]    <other_subscription_name>             <other_subscription_id>               canonical.com
+
+...
+
+```
+
+In the prompt, select the subscription id you would like to connect the controller to, and store the id 
+as it will be needed in the next step when bootstrapping the controller. 
+
+### Bootstrap Juju controller on Azure
+
+First, you need to add a set of credentials to your Juju client:
+
+```shell
+juju add-credentials azure
+```
+
+This will start a script that will help you set up the credentials, where you will be asked:
+
+* `credential-name`: fill it with a sensible name that will help you identify the credential set, say `<CREDENTIAL_NAME>`
+* `region`: select any default region is more convenient for you to deploy your controller and applications. Note that credentials are nevertheless not region specific
+* `auth type`: select `interactive`, which is the recommended way to authenticate to Azure using Juju
+* `subscription_id`: use the value `<subscription_id>` taken in the previous step
+* `application_name`: please generate a random string to avoid collision with other users or applications
+* `role-definition-name`: please generate a random string to avoid collision with other users or applications, and store it as `<AZURE_ROLE>`
```

Comment:
```suggestion
* `credential-name` — a sensible name that will help you identify the credential set, say `<CREDENTIAL_NAME>`
* `region` — a default region that is most convenient to deploy your controller and applications. Note that credentials are not region-specific
* `auth type` — authentication type. Select `interactive`, which is the recommended way to authenticate to Azure using Juju
* `subscription_id` — the value `<subscription_id>` taken in the previous step
* `application_name` — any unique string to avoid collision with other users or applications
* `role-definition-name` — any unique string to avoid collision with other users or applications, and store it as `<AZURE_ROLE>`
```

Reply:
Uhm, I had a look at the styleguide, e.g. [here](https://docs.ubuntu.com/styleguide/en/#headings), and there are examples when rather using the ":" over the `-`. 

Why would you use `-`?

Reply:
I don't see any directly relevant examples in our style guide. This is a definition/glossary-style fragment. I'm used to using either m-dash or n-dash for those. I don't mind using colons, but they are harder to see and navigate in this case. Here is some relevant info with references to The Chicago Manual of Style: https://english.stackexchange.com/questions/459638/should-i-use-a-hyphen-an-en-dash-or-an-em-dash-to-define-or-introduce-a-word
If you have a strong preference for colons here, I'm OK with that as long as we will do it consistently everywhere. So dash was my preference so far.

Reply:
> I don't see any directly relevant examples in our style guide.

In the "Dates" section, but also the document has several " * Use: ". 

Anyhow, I don't have any strong preference, and I agree for targeting consistency. Actually I found that in `h-cluster-migration.md` we use dashes, so let's go with that.